src/app/models/indicator.model.ts

```diff
@@ -10,6 +10,7 @@ export class Indicator {
     available_units?: string[];
     default_units?: string;
     parameters?: IndicatorParameter[];
+    thresholdIndicator: boolean;
```

Comment:
I don't think this is an extensible approach. The Indicator class (which should actually be an interface) is intended to match the Indicator object returned from the API, and now it contains additional items that describe an instance's state, rather than its properties. It also requires extra logic in the `list()` method of the Indicator service that needs to be duplicated anwhere an Indicator is returned from the API, such as in Indicator.getData, which returns an indicator as one of the top level objects in its response.

Since this is a "derived" property of an Indicator, meaning that its state can be calculated from another property of indicator, it makes more sense as a method on the Indicator class, or in an indicator utils service that has a method such as `isThresholdIndicator(indicator: Indicator) => boolean`. The first approach is more OOP-like, but has the same disadvantage as the solution here in that it requires `new`ing each indicator object from all the API responses so that you get instances of the class rather than just an object that matches the Indicator interface.

We haven't done enough work with TypeScript/Angular to have a good sense of how to approach these types of problems unfortunately, so I don't have a sense of which is best. But I think either are preferable to having significant logic required to properly create these Indicator instances that exists outside of the Indicator model file.

Another option for the method that tells what "type" of indicator a given indicator is, rather than having a boolean method for each type, might be something like `getType(indicator: Indicator) => IndicatorTypeEnum`.